### PR TITLE
Update SDK template to conform with Track 2 client library shape

### PR DESCRIPTION
**Pull request by srnagar**
Minor updates the SDK template to update the package name, add module-info.java and use JUnit 5. This template is used as a starting point for partner teams to clone and start working on new client libraries. So, setting this up correctly is going to make it easy for getting started.

### CODE REVIEW

File: sdk/template/azure-sdk-template/src/main/java/com/azure/sdk/template/Hello.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-package com.microsoft.azure.template;
+package com.azure.sdk.template;
```
Comment (samvaity) [inline sdk/template/azure-sdk-template/src/main/java/com/azure/sdk/template/Hello.java:3]: It also seems to be following the old naming for directory structure.
Should we consider updating that too?
image

File: sdk/template/azure-sdk-template/pom.xml [hunk 2]
```diff
@@ -34,9 +34,21 @@
       <version>1.6.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
```
Comment (samvaity) [inline sdk/template/azure-sdk-template/pom.xml:34]: nit: The Readme.md also seems to be outdated for the current Track 2

File: sdk/template/azure-sdk-template/src/main/java/com/azure/sdk/template/Hello.java (hunk 1 above)
Comment (srnagar) [reply sdk/template/azure-sdk-template/src/main/java/com/azure/sdk/template/Hello.java:3]: The directories are updated too. `src/main/java/com/azure/sdk/template/Hello.java`

File: sdk/template/azure-sdk-template/pom.xml (hunk 2 above)
Comment (srnagar) [reply sdk/template/azure-sdk-template/pom.xml:34]: I'll leave the Readme.md as-is for now. The readme has some guidance that can be useful for partners.

**Pull request merged**